maestro-ios/src/main/java/ios/xctest/XCTestIOSDevice.kt

```diff
@@ -40,7 +41,31 @@ class XCTestIOSDevice(
     }
 
     override fun deviceInfo(): Result<DeviceInfo, Throwable> {
-        error("Not supported")
+        return runCatching {
+            client.deviceInfo().use { response ->
+                val body = response.body?.bytes()?.let { String(it) }
+
+                if (!response.isSuccessful) {
+                    val message = "${response.code} ${response.message} - $body"
+                    logger.info("Device info failed: $message")
+                    throw UnknownFailure(message)
+                }
+
+                body ?: throw UnknownFailure("Error: response body missing")
+
+                val responseBody = mapper.readValue(body, DeviceInfoResponse::class.java)
+                logger.info("Device info $responseBody")
+
+                // Warning: assuming pixels == points
```

Comment:
technically we can get them as well by multiplying on `UIScreen.main.scale`

Reply:
hmm, just tested , worked for me

Reply:
You are correct sir! Thanks for testing! I'll update the PR accordingly